### ISSUE TITLE
Android "buildToolsVersion is not specified"

### ISSUE THREAD

**Issue by Noitidart** [prscX/react-native-popover-menu#5]
I did:

```
npm i react-native-popover-menu
react-native link react-native-popover-menu
react-native start
react-native run-android
```

But then the run-android command ends with:

```
PS C:\Users\Mercurius\Documents\GitHub\did> react-native run-android
Scanning folders for symlinks in C:\Users\Mercurius\Documents\GitHub\did\node_modules (47ms)
JS server already running.
Building and installing the app on the device (cd android && gradlew.bat installDebug)...

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':app'.
> Could not resolve all dependencies for configuration ':app:_debugApk'.
   > A problem occurred configuring project ':react-native-popover-menu'.
      > buildToolsVersion is not specified.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED
```

**Comment by prscX** [comment]
Thanks @Noitidart for raising the issue.

Basically `buildToolsVersion` is missing from library. Can you please try `buildToolsVersion` as `27.0.3` in rn-popover-menu `build.gradle` file and run it once.

I was under impression that it should pick `buildToolsVersion` from your app and build the same. If it works can you please raise a PR for the same.

Thanks
Pranav

**Comment by Noitidart** [comment]
On it right away. Will report back how it goes.

If I change build tools version, when I compile my app, will it work down to Android 4.1 (which is minimum android RN supports)?

**Comment by Noitidart** [comment]
So I went into build.gradle here - https://github.com/prscX/react-native-popover-menu/blob/master/android/build.gradle#L16

I changed it to:
```
android {
    compileSdkVersion 27
    buildToolsVersion "27.0.3"
```
Just added that third line.



However I get this error when I run react-native run-android:

In short I think the error is `com.android.support:appcompat-v7:27.0.2`. Here is full:

```
PS C:\Users\Mercurius\Documents\GitHub\did> react-native run-android
Scanning folders for symlinks in C:\Users\Mercurius\Documents\GitHub\did\node_modules (46ms)
JS server already running.
Building and installing the app on the device (cd android && gradlew.bat installDebug)...
File C:\Users\Mercurius\.android\repositories.cfg could not be loaded.
Checking the license for package Android SDK Build-Tools 27.0.3 in C:\Users\Mercurius\AppData\Local\Android\sdk\licenses
License for package Android SDK Build-Tools 27.0.3 accepted.
Preparing "Install Android SDK Build-Tools 27.0.3".
"Install Android SDK Build-Tools 27.0.3" ready.
Finishing "Install Android SDK Build-Tools 27.0.3"
Installing Android SDK Build-Tools 27.0.3 in C:\Users\Mercurius\AppData\Local\Android\sdk\build-tools\27.0.3
"Install Android SDK Build-Tools 27.0.3" complete.
Download https://jcenter.bintray.com/com/github/zawadz88/materialpopupmenu/material-popup-menu/1.2.0/material-popup-menu-1.2.0.pom
Download https://jcenter.bintray.com/org/jetbrains/kotlin/kotlin-stdlib-jre7/1.2.20/kotlin-stdlib-jre7-1.2.20.pom
Download https://jcenter.bintray.com/org/jetbrains/kotlin/kotlin-stdlib/1.2.20/kotlin-stdlib-1.2.20.pom
Download https://jcenter.bintray.com/org/jetbrains/annotations/13.0/annotations-13.0.pom

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':app'.
> Could not resolve all dependencies for configuration ':app:_debugApk'.
   > A problem occurred configuring project ':react-native-popover-menu'.
      > Could not resolve all dependencies for configuration ':react-native-popover-menu:_debugPublishCopy'.
         > Could not find com.android.support:appcompat-v7:27.0.2.
           Searched in the following locations:
               file:/C:/Users/Mercurius/AppData/Local/Android/sdk/extras/android/m2repository/com/android/support/appcompat-v7/27.0.2/appcompat-v7-27.0.2.pom
               file:/C:/Users/Mercurius/AppData/Local/Android/sdk/extras/android/m2repository/com/android/support/appcompat-v7/27.0.2/appcompat-v7-27.0.2.jar
               file:/C:/Users/Mercurius/Documents/GitHub/did/android/sdk-manager/com/android/support/appcompat-v7/27.0.2/appcompat-v7-27.0.2.jar
           Required by:
               did:react-native-popover-menu:unspecified > com.facebook.react:react-native:0.53.0
         > Could not find com.android.support:appcompat-v7:27.0.2.
           Searched in the following locations:
               file:/C:/Users/Mercurius/AppData/Local/Android/sdk/extras/android/m2repository/com/android/support/appcompat-v7/27.0.2/appcompat-v7-27.0.2.pom
               file:/C:/Users/Mercurius/AppData/Local/Android/sdk/extras/android/m2repository/com/android/support/appcompat-v7/27.0.2/appcompat-v7-27.0.2.jar
               file:/C:/Users/Mercurius/Documents/GitHub/did/android/sdk-manager/com/android/support/appcompat-v7/27.0.2/appcompat-v7-27.0.2.jar
           Required by:
               did:react-native-popover-menu:unspecified > com.github.zawadz88.materialpopupmenu:material-popup-menu:1.2.0
         > Could not find com.android.support:recyclerview-v7:27.0.2.
           Searched in the following locations:
               file:/C:/Users/Mercurius/AppData/Local/Android/sdk/extras/android/m2repository/com/android/support/recyclerview-v7/27.0.2/recyclerview-v7-27.0.2.pom
               file:/C:/Users/Mercurius/AppData/Local/Android/sdk/extras/android/m2repository/com/android/support/recyclerview-v7/27.0.2/recyclerview-v7-27.0.2.jar
               file:/C:/Users/Mercurius/Documents/GitHub/did/android/sdk-manager/com/android/support/recyclerview-v7/27.0.2/recyclerview-v7-27.0.2.jar
           Required by:
               did:react-native-popover-menu:unspecified > com.github.zawadz88.materialpopupmenu:material-popup-menu:1.2.0
```

**Comment by Noitidart** [comment]
I noticed some lines that differed in the example build.gradle

There is `google()` here:

* https://github.com/prscX/react-native-popover-menu/blob/master/Example/android/build.gradle#L6
* https://github.com/prscX/react-native-popover-menu/blob/master/Example/android/build.gradle#L24

And then here it is `3.0.1`:

* https://github.com/prscX/react-native-popover-menu/blob/master/Example/android/build.gradle#L9

Mine doesn't have `google()` and is `classpath 'com.android.tools.build:gradle:2.2.3'` (this is a fresh react-native init project just made it the other day).

So I added google() and changed to 3.0.1 but its giving me error:

```
Build file 'C:\Users\Mercurius\Documents\GitHub\did\android\build.gradle' line: 6

* What went wrong:
A problem occurred evaluating root project 'did'.
> Could not find method google() for arguments [] on repository container.
```

**Comment by danielang** [comment]
I've solved this problem with the following modification in my android/build.gradle:
```gradle
subprojects {
    afterEvaluate {
        project -> if (project.hasProperty("android")) {
            android {
                compileSdkVersion 26
                buildToolsVersion '26.0.2'
            }
        }
    }
}
```

And update my android/app/build.gradle with the following:
```gradle
android {
    compileSdkVersion 26
    buildToolsVersion "26.0.2"

   ...other stuff
}
```

**Comment by Noitidart** [comment]
Thanks @danielang! I will test this on the weekend when I get back, and report back.

**Comment by prscX** [comment]
Thanks @danielang for sharing the fix.

In library `build.gradle` file `buildToolsVersion` is not specified, should we include it?

Thanks
Pranav

**Comment by danielang** [comment]
Yes, I think so. Every other react-native library I saw provides this entries in their `build.gradle`file. For example:  
* [Firebase](https://github.com/invertase/react-native-firebase/blob/master/android/build.gradle)
* [RN-Device-Info](https://github.com/invertase/react-native-firebase/blob/master/android/build.gradle)
* [RN-Gesture-Handler](https://github.com/kmagiera/react-native-gesture-handler/blob/master/android/build.gradle)

**Comment by prscX** [comment]
Hi @danielang + @Noitidart, 

I have added the `buildToolsVersion` in library `build.gradle`. I am working on couple of other issues as well, once those are resolved I will publish the latest version.

Thanks
Pranav

**Comment by prscX** [comment]
Hi @danielang + @Noitidart, I am closing this issue since it was fixed the latest version and wanted to focus the open items.

Let me know in case you face any issue.

Thanks
Pranav

**Comment by Noitidart** [comment]
Thank you @prscX !! Will try when latest is on npm

**Comment by prscX** [comment]
Hi @Noitidart, Latest version is already publish on npm, you can try by installing latest v0.0.4.

Thanks
Pranav

**Comment by Noitidart** [comment]
Thanks @prscX! I'm trying it now. :)

**Comment by Noitidart** [comment]
Darn. I did `react-native init onesignal` (just dummy name for new project). Then npm i then react-native-link and then react-native start and react-native run-android and I am getting same error:

```
* What went wrong:
A problem occurred configuring project ':app'.
> Could not resolve all dependencies for configuration ':app:_debugApk'.
   > A problem occurred configuring project ':react-native-popover-menu'.
      > buildToolsVersion is not specified.
```

:(

**Comment by prscX** [comment]
My sincerely apologise, not sure how this got missed.

I have released a new V0.0.6, can you please download and verify the same. 

Please make sure that your app SDK is 27 >. Let me know in case you still face any issues.

Thanks
Pranav
